src/modules/localStorage.js

```diff
@@ -0,0 +1,79 @@
+const entryTask = document.getElementById('newTask');
+
+const storageAvailable = (type) => {
+  try {
+    const storage = window[type];
+    const x = '__storage_test__';
+    storage.setItem(x, x);
+    storage.removeItem(x);
+    return true;
+  } catch (e) {
+    return e instanceof DOMException && (
+    // everything except Firefox
+      e.code === 22
+    // Firefox
+    || e.code === 1014
+    // test name field too, because code might not be present
+    // everything except Firefox
+    || e.name === 'QuotaExceededError'
+    // Firefox
+    || e.name === 'NS_ERROR_DOM_QUOTA_REACHED');
+  }
+};
+
+export const savedata = () => {
+  const taskToDo = JSON.parse(localStorage.getItem('toDoList')) || [];
+  if (storageAvailable('localStorage')) {
+    const obj = {
+      description: `${entryTask.value}`,
+      completed: false,
+      index: JSON.parse(`${taskToDo.length}`),
+    };
+    taskToDo.push(obj);
+    localStorage.setItem('toDoList', JSON.stringify(taskToDo));
+  }
+};
+
+export const retrivedata = () => {
+  const taskToDo = JSON.parse(localStorage.getItem('toDoList')) || [];
+  const list = document.getElementById('uList');
+  document.querySelectorAll('.task-style').forEach((t) => t.remove());
+
+  taskToDo.forEach((tsk) => {
+    const task = document.createElement('li');
+    task.innerHTML = `<span><input type="checkbox" id= "id-${tsk.index}">
+    <label for= "id-${tsk.index}" id="lb-${tsk.index}">${tsk.description}</label></span>
+    <i class= "menu" id="${tsk.index}">__</i>`;
+    task.classList.add('task-style');
+    task.id = `li-${tsk.index}`;
+    list.appendChild(task);
+  });
+
+  const checkBox = document.querySelectorAll('input[type=checkbox]');
+
+  checkBox.forEach((box) => box.addEventListener('click', () => {
+    const num = [...`${box.id}`].splice(3).pop();
+    if (box.checked) {
+      taskToDo[num].completed = true;
+    } else {
+      taskToDo[num].completed = false;
+    }
+    localStorage.setItem('toDoList', JSON.stringify(taskToDo));
+  }));
```

Comment:
- [ ] (Optional) It would be nice to use a Javascript dataset to give value to the checkboxes. When you loop using `forEach` you can use the second index parameter to check if it is equal t the value of the dataset attribute. If it is, you can set the value of the **_completed_** to `true`.

You can read this [article](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset) to get a clearer understanding of how to use datasets.

Reply:
I couldn't implement this because I was running against time, but I'm looking for the opportunity to use it! 😃 👍🏼